### PR TITLE
Add value userType to sharees api 

### LINKED ISSUE #38422: [QA] guest user email is shown as normal user, when used for the second time
Seen with owncloud server 10.6.0 and guests app 0.9.0 

* Enter a yet unknown email address in the share dialog.
* A choice appears for "federated" or "guest". OK

image

* Click the guest. ownCloud creates a user entry for the guest. 
* Can be verified in Admin Users or e.g. metrics overview:
image

* Enter the same email address again for another share. The choice now is "user" and "Federated". BAD
image

Expected behaviour: guest users are always flagged as "guest".

### PR DESCRIPTION
**Pull request by AlexAndBear**
## Related Issue

- Fixes https://github.com/owncloud/core/issues/38422

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Review by mrow4a — COMMENTED** [review]
can api return „userType” instead? is this possible? this would make it more generic

**Comment by JanAckermann** [comment]
> can api return „userType” instead? is this possible? this would make it more generic

Of course, we can do this, are there more types than 'user' and 'guest'?

**Comment by mrow4a** [comment]
> > can api return „userType” instead? is this possible? this would make it more generic
> 
> Of course, we can do this, are there more types than 'user' and 'guest'?

needs to be investigated, best if returns some ID / CONST defined in core.

**Comment by JanAckermann** [comment]
@mrow4a ok updated in that favor, might have quick look? if you give me green light, I will update the tests

File: lib/public/User.php [hunk 1]
```diff
@@ -44,7 +44,7 @@
  * about the currently logged in user and the permissions for example
  * @since 5.0.0
  */
-class User {
+class User extends \OC\User\Constants{
```
Comment (mrow4a) [inline lib/public/User.php]: public/User using private/User ? Constanst should be public, and not private implementation

**Review by mrow4a — CHANGES_REQUESTED** [review]
generally makes sense, one small detail to be fixed

File: lib/public/User.php (hunk 1 above)
Comment (JanAckermann) [reply lib/public/User.php]: I took the share implementation as boilerplate

**Comment by JanAckermann** [comment]
@mrow4a Tests passed, approve?

**Review by mrow4a — APPROVED** [review]

**Pull request merged**